### ISSUE TITLE
input requirements of flashselfattn

### ISSUE THREAD

**Issue by taoye9** [tridao/zoo#1]
Hi, i‘m trying to use flashattn on cases where q,k ,v are of different values.

could someone confirm whether  flashSelfAttn forward inputs (q, k, v)are a same tensor/values are same?

as the world self attn usually implies q,k,v are a same tensor.

code is here:
https://github.com/tridao/zoo/blob/14175433dc55524e7f436303e5bb4c3d1c8a802a/src/models/modules/mha.py?from_wecom=1#L24-L60

**Comment by tridao** [comment]
Yes. There's also a cross attention version where q is separate by k, v are stacked into 1 tensor. Or you can call flash_attn_unpadded_func which takes in separate q, k, v tensors.
